okonomiyaki/versions/semver.py

```diff
@@ -0,0 +1,197 @@
+import re
+
+from ..utils.py3compat import long
+
+
+_SEMVER_R = re.compile("""\
+    (\d+)\.(\d+)\.(\d+)   # main part
+    (-[0-9a-zA-Z-\.]+)?
+    (\+[0-9a-zA-Z-\.]+)?
+    $
+""", flags=re.VERBOSE)
+
+
+_PART_R = re.compile("[0-9a-zA-Z-]+")
+
+
+def _ensure_no_leading_zero(value, name):
+    if len(value) > 1 and value.startswith("0"):
+        msg = "{0} number cannot have leading 0: {1!r}".format(name, value)
+        raise ValueError(msg)
+
+
+def _parse_pre_release(s):
+    if s is not None:
+        return tuple(part for part in s[1:].split("."))
```

Comment:
This is very opaque; it took a while to realize that the leading `-` or `+` is included in the input here. It is also not clear why we store just the "raw" tuple rather than a "parsed" `_PrereleaseParts` instance.